src/xenia/kernel/xam/user_profile.cc

```diff
@@ -10,25 +10,53 @@
 #include "xenia/kernel/xam/user_profile.h"
 
 #include <sstream>
+#include <xhash>
 
 #include "third_party/fmt/include/fmt/format.h"
 #include "xenia/kernel/kernel_state.h"
 #include "xenia/kernel/util/shim_utils.h"
 
+DEFINE_string(gamertag_0, "User_0", "The name of your user profile.",
+              "Profile");
+DEFINE_string(gamertag_1, "User_1", "The name of your user profile.",
+              "Profile");
+DEFINE_string(gamertag_2, "User_2", "The name of your user profile.",
+              "Profile");
+DEFINE_string(gamertag_3, "User_3", "The name of your user profile.",
+              "Profile");
+
 namespace xe {
 namespace kernel {
 namespace xam {
 
 UserProfile::UserProfile(uint8_t index) {
+  const std::vector<std::string> username_list = {
+      cvars::gamertag_0,
+      cvars::gamertag_1,
+      cvars::gamertag_2,
+      cvars::gamertag_3,
+  };
+
+  std::string current_username = username_list[index];
+  if (current_username.empty()) {
+    current_username = "User_" + std::to_string(index);
+  }
+
+  // TODO(Gliniak): Username sanitizer
+  //
   // 58410A1F checks the user XUID against a mask of 0x00C0000000000000 (3<<54),
   // if non-zero, it prevents the user from playing the game.
   // "You do not have permissions to perform this operation."
-  xuid_ = 0xB13EBABEBABEBABE + index;
-  name_ = "User";
-  if (index) {
-    name_ = "User_" + std::to_string(index);
-  }
 
+  // Get assigned username to current index from config
+  // Then use lower 4 bytes based on player name for unique xuid
+  size_t hash = std::hash<std::string>{}(current_username);
+
+  xuid_ = 0xB13E000000000000 + (hash & 0xFFFFFFFF);
```

Comment:
If the plan is to change XUID here, we should probably sort out the upper part here. I don't think real Xbox devices use 0xB13E.
An offline XUID is one that starts with 0xE, so probably could go for 0xE000. `hdd1:\content` is probably the place to look if you want to see how a real Xbox does it.

Reply:
You're right. All should start from 0xE000. I will change it soon.